### PR TITLE
implemented hourly grouped data

### PR DESCRIPTION
**Pull request by thanosbnt**

### CODE REVIEW

File: Analytics/resources/request_grouped.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
+import pandas as pd
+import json
+import numpy as np
+from models.sensor import Sensor
+from models.location import Location
+
+"""
+This function groupes sensor data having different temporal frequencies by grouping at
```
Comment (sassalley) [inline Analytics/resources/request_grouped.py]: Small typo - `This function groups`

File: Analytics/resources/request_grouped.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+		harm_df = harm_df.set_index('Timestamp')
+		harm_df.Value = harm_df.Value.astype(float)
+
+		if not per_sensor:
+			### Using the median to preserve the data type during groupby aggregation
+			harm_df = harm_df.groupby([pd.Grouper(freq='1H'), 'Attribute_Name']).median()
+			harm_df.reset_index(inplace=True)
+
+		else:
+			harm_df = harm_df.groupby([pd.Grouper(freq='1H'), 'Attribute_Name', 'Sensor_id']).median()
```
Comment (sassalley) [inline Analytics/resources/request_grouped.py]: Can we enable the API to specify `freq`? 

E.g. by adding:

- Weekly:`1W`
- Daily:`1D`
- Hourly:`1H` (as already implemented)
- Minutely:`1T`

File: Analytics/resources/request_for_data.py [hunk 3]
```diff
@@ -35,11 +38,11 @@
 		{URL}?sensor='<name1>,<name2>' // To retrieve multiple records
 		{URL}?attributedata='<name-of-attribute>&limit=60&offset=60' // Retrieve records but increase limit and skip 60
 		{URL}?attributedata='<name1><name2>&limit=60&offset=60&fromdate=2018-11-22&todate=2018-11-24'
-
+		{URL}?attributedata='<name1><name2>&limit=1000&grouped=True&per_sensor=True // Retrieves records and groups the data at hourly intervals
```
[the remark this answers is not in the document]
Comment (thanosbnt) [reply Analytics/resources/request_for_data.py]: yes this is correct. I'll make the update. It also needs examples of the harmonising_method which i will add as well

File: Analytics/resources/request_grouped.py (hunk 1 above)
Comment (thanosbnt) [reply Analytics/resources/request_grouped.py]: good catch. Thanks!

File: Analytics/resources/request_grouped.py (hunk 2 above)
Comment (thanosbnt) [reply Analytics/resources/request_grouped.py]: Yes that should be straighforward. I'll implement it

**Comment by sassalley** [comment]
Looks good now! Need to add tests as discussed.

**Pull request merged**